src/Octoshift/Services/OctoLogger.cs

```diff
@@ -18,6 +19,7 @@ internal static class LogLevel
 public class OctoLogger
 {
     public virtual bool Verbose { get; set; }
+    private readonly HashSet<Regex> _redactionPatterns = new();
```

Comment:
I would keep all patterns as strings right here in the `OctoLogger` class instead of registering them via a public method, so in this case we will have:
```c#
private readonly List<string> _recationPatterns = new()
{
    "your regex pattern here"
};
```